### ISSUE TITLE
Error: Python shared library not found, Python bindings not loaded

### ISSUE THREAD

**Issue by christopherjlawrence** [quanteda/spacyr#213]
Hello, hope you all can help with this. 

### Issue

I've got both the spacyr and reticulate packages loaded into my RStudio project. I also installed miniconda on my computer. However, when I run:

`spacy_initialize()`

I get:

> Error: Python shared library not found, Python bindings not loaded.
> Use reticulate::install_miniconda() if you'd like to install a Miniconda Python environment.

I don't understand this since I have miniconda installed on my computer. R apparently doesn't think the Python I have is shared, I guess. 

sessionInfo() included below.

### sessionInfo()

> R version 4.1.1 (2021-08-10)
> Platform: x86_64-apple-darwin17.0 (64-bit)
> Running under: macOS Big Sur 11.6
> 
> Matrix products: default
> LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
> 
> locale:
>   [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
> 
> attached base packages:
>   [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
>   [1] reticulate_1.22   spacyr_1.2.1      textrecipes_0.4.1 recipes_0.1.17   
> [5] cleanNLP_3.0.3    hunspell_3.0.1    SnowballC_0.7.0   gt_0.3.1         
> [9] magrittr_2.0.1    rmarkdown_2.11    htmltools_0.5.2   rlang_0.4.12     
> [13] stopwords_2.2     quanteda_3.1.0    text2vec_0.6      corpus_0.10.2    
> [17] bench_1.1.1       jiebaR_0.11       jiebaRD_0.1       janeaustenr_0.1.5
> [21] hcandersenr_0.2.0 tidytext_0.3.2    forcats_0.5.1     stringr_1.4.0    
> [25] dplyr_1.0.7       purrr_0.3.4       readr_2.0.2       tidyr_1.1.4      
> [29] tibble_3.1.5      ggplot2_3.3.5     tidyverse_1.3.1   tokenizers_0.2.1 
> 
> loaded via a namespace (and not attached):
>   [1] colorspace_2.0-2     ellipsis_0.3.2       class_7.3-19        
> [4] rprojroot_2.0.2      fs_1.5.0             rstudioapi_0.13     
> [7] listenv_0.8.0        remotes_2.4.1        prodlim_2019.11.13  
> [10] fansi_0.5.0          lubridate_1.8.0      xml2_1.3.2          
> [13] codetools_0.2-18     splines_4.1.1        cachem_1.0.6        
> [16] rsparse_0.4.0        knitr_1.36           pkgload_1.2.3       
> [19] mlapi_0.1.0          jsonlite_1.7.2       RhpcBLASctl_0.21-247
> [22] broom_0.7.9          dbplyr_2.1.1         png_0.1-7           
> [25] compiler_4.1.1       httr_1.4.2           backports_1.2.1     
> [28] assertthat_0.2.1     Matrix_1.3-4         fastmap_1.1.0       
> [31] cli_3.0.1            prettyunits_1.1.1    tools_4.1.1         
> [34] gtable_0.3.0         glue_1.4.2           float_0.2-6         
> [37] fastmatch_1.1-3      Rcpp_1.0.7           cellranger_1.1.0    
> [40] vctrs_0.3.8          timeDate_3043.102    xfun_0.27           
> [43] gower_0.2.2          globals_0.14.0       ps_1.6.0            
> [46] testthat_3.1.0       rvest_1.0.2          lifecycle_1.0.1     
> [49] devtools_2.4.2       future_1.22.1        MASS_7.3-54         
> [52] scales_1.1.1         ipred_0.9-12         lgr_0.4.3           
> [55] hms_1.1.1            parallel_4.1.1       yaml_2.2.1          
> [58] memoise_2.0.0        rpart_4.1-15         stringi_1.7.5       
> [61] desc_1.4.0           pkgbuild_1.2.0       lava_1.6.10         
> [64] pkgconfig_2.0.3      evaluate_0.14        lattice_0.20-45     
> [67] processx_3.5.2       tidyselect_1.1.1     parallelly_1.28.1   
> [70] R6_2.5.1             generics_0.1.0       DBI_1.1.1           
> [73] pillar_1.6.4         haven_2.4.3          withr_2.4.2         
> [76] nnet_7.3-16          survival_3.2-13      future.apply_1.8.1  
> [79] modelr_0.1.8         crayon_1.4.1         utf8_1.2.2          
> [82] tzdb_0.1.2           usethis_2.1.0        grid_4.1.1          
> [85] readxl_1.3.1         data.table_1.14.2    callr_3.7.0         
> [88] reprex_2.0.1         digest_0.6.28        RcppParallel_5.1.4  
> [91] munsell_0.5.0        sessioninfo_1.1.1

**Comment by christopherjlawrence** [comment]
I'm running:
```
library(reticulate)
library(spacyr)
use_virtualenv("spacy_condaenv")
spacy_initialize()
```
Only to get:
```
Found 'spacy_condaenv'. spacyr will use this environment
Error: Python shared library not found, Python bindings not loaded.
Use reticulate::install_miniconda() if you'd like to install a Miniconda Python environment.
```

Why doesn't it like my Python? I downloaded miniconda and didn't mess with anything during the install.

**Comment by christopherjlawrence** [comment]
I added this on StackOverflow and wanted to add here. 

Someone provided me with a solution.

Have loaded: reticulate and spacyr packages.

Turns out that the reticulate package doesn't follow the standard way of indicating which version of Python you'd like to use, at least on a Mac OS, which has a system-level default that R wants to use.

The workaround is to set the system environment variable to tell reticulate to the attach the desired conda environment.

Sys.setenv( RETICULATE_PYTHON = with(conda_list(), python[ name == "spacy_condaenv" ]))

Next, run spacy_initialize(). Should work now.

Going forward, you'll need to put the Sys.setenv code (as written above) at the top of your script so that R sets the system environment to your liking right when you start a session.
